cmd/agent/main_linux.go

```diff
@@ -0,0 +1,18 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package main
+
+import (
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+func setProcessName(process string) error {
+	processName := make([]byte, len(process)+1)
+	copy(processName, process)
+	return unix.Prctl(unix.PR_SET_NAME, uintptr(unsafe.Pointer(&processName[0])), 0, 0, 0)
```

Comment:
My understanding is that PR_SET_NAME sets the name of the thread not of the process. Since we are in go, there is no real way to do this in the main thread/all threads..

Reply:
I would suggest maybe using https://pkg.go.dev/syscall#AllThreadsSyscall ?

Reply:
But it looks like it doesn't support binaries with cgo..